code/modules/clothing/spacesuits/rig/rig.dm

```diff
@@ -771,6 +784,10 @@
 
 	if(!usr.incapacitated())
 		combat_mode = !combat_mode
+		var/obj/item/clothing/head/helmet/space/rig/syndi/H = helmet
+		if(istype(H))
+			if(H.combat_mode != combat_mode)
+				H.toggle()
```

Comment:
```suggestion
		if(H?.combat_mode != combat_mode)
			H.toggle()
```
мб лучше так?

Reply:
Я сделал небольшие замеры.

У меня участвовали три записи. Абсолютный фаворит по медленности это с истайпом. Самый быстрый это c `?.` . Так что вот, да. 

```
if (A && A.foo)
===
if (istype(A) && A.foo)
===
if (A?.foo)
```

Ссылка на гист для проверки
https://gist.github.com/KIBORG04/d878be4b1f43796fe49dd76d8f6c9e57

Ну или с кодербуса
https://discord.com/channels/484170914754330625/487268744419344384/813880305362403378

Reply:
Истайп же проверяет на тип, в то время как ```?.``` Смотрит, чтобы переменная, из которой хотят взять значение, не было пустым. В данном плане, если нужно также убедиться что тип верен, заменять на ?. совсем не корректно.

Reply:
А зачем в данных ситуация проверять на `istype(A, A.type)`? (Как я понял из доков, то истайп без второго аргумента делает именно так)

Reply:
> А зачем в данных ситуация проверять на istype(A, A.type)? (Как я понял из доков, то истайп без второго аргумента делает именно так)

If no type is specified and a variable was passed in as the first argument, it will default to the declared type of the variable. 

Он проверяет, соответствует ли тип значения объявленному типу переменной
В данном случае это равнозначно, если бы написать
```
istype(H, /obj/item/clothing/head/helmet/space/rig/syndi)
```